src/Fields/interpolate.jl

```diff
@@ -77,7 +77,7 @@ end
     loc = @inbounds locs[1]
      Tx = topology(grid, 1)()
      Nx = length(loc, Tx, grid.Nx)
-     xn = xnodes(grid, locs...)
+     xn = xnodes(grid, locs..., with_halos=true)
```

Comment:
not sure but you may want to make `with_halos` a positional argument for this to be efficient in a kernel

something like

```julia
xnodes(grid, lx, ly, lz, with_halos)
```

plus

```julia
xnodes(grid, lx, ly, lz; with_halos=false) = xnodes(grid, lx, ly, lz, with_halos)
```

to preserve default behavior

Reply:
sounds good, no keyword arguments in kernels is definitely preferred